qiskit_experiments/base_experiment.py

```diff
@@ -99,17 +112,22 @@ def run(
         Returns:
             The experiment data object.
         """
-        # Create new experiment data
         if experiment_data is None:
-            experiment_data = self.__experiment_data__(self, backend=backend)
-
-        # Generate and transpile circuits
-        circuits = transpile(self.circuits(backend), backend, **self.transpile_options.__dict__)
+            # Create new experiment data
+            experiment_data = self.__experiment_data__(experiment=self, backend=backend)
+        else:
+            # Override experiment metadata
```

Comment:
Not sure if override is the correct thing to do. Need to better understand the use case of running an experiment again and adding the result to the existing experiment data. Do we override old analysis results, that are stored in this experiment data? If not, then have these analysis results recorded the parameters they were running with? Possibly we should raise an exception if the old and new options don't match.

Reply:
(instead of overriding the old options)

Reply:
Yes, adding to an existing data container is still not clear how well it will work with the result DB so I'm not entirely sure what is best to do here. I thought about raising an exception, and maybe that is better. One question is how strict should the checks be ie if experiment type or qubits changes you should raise, but if experiment options or change probably not since changing those may be the main reason for adding more data. In that case im not sure what good way is to handle overriding old options with new options.

Reply:
I agree with @yaelbh . In the example of T1, if we run the first experiment with `delay=[1,2,3]` and second one with `[4, 5, 6]` the delay appeared in the experiment metadata should be `[1, 2, 3, 4, 5, 6]` otherwise we cannot regenerate experiment data with attached metadata (this is very confusing). Other option is to keep all runtime configuration as `set`, or not storing runtime options.

Reply:
I guess metadata could be a list of dict instead of dict and each time you run it appends a new metadata dict to the list so you have all of them.

Reply:
I changed it to add a field `"job_options": {}` which on execution will store a dict of all the various options dicts keyed by the job ID. This way no options from previous runs will be overriden.